test/common/http/common.h

```diff
@@ -31,7 +32,7 @@ class CodecClientForTest : public Http::CodecClient {
   }
 
   void raiseGoAway() { onGoAway(); }
-
+  Event::TimerPtr idleTimer() { return std::move(idle_timer_); }
```

Comment:
`Event::Timer* idleTimer() { return idle_timer_.get(); }`

Reply:
if I change this, it never returns nullptr even after the reset call and the following check in codec_client_test fails.
  EXPECT_EQ(client_->idleTimer(), nullptr);
So reverted it like this. Let me know if there is any other better way for this check.

Reply:
Something doesn't make sense. The move call to return the local variable is not correct. If the above is returning nullptr I think it's likely doing so by accident somehow. If you are trying to look at the member variable state, you should do it by pointer, not move.

Another way to handle is to just remove this function entirely, and expect `disableTimer()`. This is generally what is done elsewhere. I think that would be cleaner anyway.

Reply:
ok. you are correct. I found the problem. The close method on CodecClient is not actually invoking onEvent and hence idletimer is not disabled in that case. It is a great catch. I have added that code in close method as well and now it works.